src/packagedcode/rubygems.py

```diff
@@ -647,93 +626,67 @@ def normalize(gem_data, known_fields=known_fields):
+                role='author'
+            )
+        )
 
-    # TODO: Check if we should use 'summary' instead of description
-    def __init__(self, location):
-        """
-        Initialize from the gem spec or gem file at location.
-        """
-        spec = parse_spec(location)
-        self.location = location
-        self.description = spec.get('description')
-        self.summary = spec.get('summary')
-        self.author = spec.get('author')
-        self.authors = spec.get('authors')
-        # can be a list
-        self.email = spec.get('email')
-
-
-        self.spec['licenses'] = self.map_licenses()
-        self.make_unique()
-
-    def __str__(self):
-        return '<{}: {}>'.format(self.__class__.__name__, self.location)
-
-    def make_unique(self):
-        """
-        Ensure that lists in the spec only contain unique values.
-        """
-        new_spec = {}
-        for key, value in self.spec.items:
-            if isinstance(value, list):
-                newlist = []
-                for item in value:
-                    if item not in newlist:
-                        newlist.append(item)
-                new_spec[key] = newlist
-            else:
-                new_spec[key] = value
-        return new_spec
-
-    def get_description(self):
-        """
-        Using 'description' over 'summary' unless summary contains
-        more data.
-        See http://guides.rubygems.org/specification-reference/
-        Note that it is common to see this is spec files: s.description = s.summary
-        """
-        description = self.spec.get('description', '')
-        summary = self.spec.get('summary', '')
-
-        content = description
-        # FIXME: we should join these.
-        if len(summary) > len(description):
-            content = summary
-
-        content = ' '.join(content.split())
-        return content.strip()
-
-    def get_email(self):
-        """
-        Join the list of emails as a comma-separated string.
-        """
-        email = self.spec.get('email', u'')
-        if isinstance(email, list):
-            email = u', '.join(email)
-        return email
-
-    def map_licenses(self):
-        licenses = self.spec.get('licenses', [])
-        if not isinstance(licenses, list):
-            licenses = [licenses]
-
-        mapped_licenses = []
-        for lic in licenses:
-            mapped_license = LICENSES_MAPPING.get(lic, None)
-            if mapped_license:
-                mapped_licenses.append(mapped_license)
-            else:
-                if TRACE:
-                    logger.warning('WARNING: {}: no license mapping for: "{}"'.format(self.filename, lic))
-        return mapped_licenses
+    package = RubyGem(
+        name=name,
+        version=version,
+        parties=parties,
+        homepage_url=homepage_url,
+        description=description,
+        declared_license=declared_license
+    )
+
+    dependencies = gemspec_data.get('dependencies')
```

Comment:
As suggested for go.mod, consider this instead here and elsewhere:
`dependencies = gemspec_data.get('dependencies', {}) or {}` and remove `if dependencies:`